src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
+
+  decreaseHealth(amount) {
+    this.health = Math.max(0, this.health - amount);
+
+    if (this.health === 0) {
+      this.die();
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(h) {
+    if (h.hidden || h instanceof Carnivore) {
+      return;
+    }
+    h.decreaseHealth(50);
```

Comment:
```suggestion
    if (!h.hidden && h instanceof Carnivore) {
      h.decreaseHealth(50);
    }
```
Just decrease the health only if conditions are true to avoid useless return